polyfill/test/validStrings.mjs

```diff
@@ -449,31 +448,33 @@ const comparisonItems = {
   YearMonth: ['year', 'month', 'calendar'],
   ZonedDateTime: [...dateItems, ...timeItems, 'offset', 'ianaName', 'calendar']
 };
-const plainModes = ['Date', 'DateTime', 'MonthDay', 'Time', 'YearMonth'];
+const plainModes = ['Instant', 'Date', 'DateTime', 'MonthDay', 'Time', 'YearMonth'];
 
-const mode = 'Instant';
-
-for (let count = 0; count < 1000; count++) {
-  let generatedData, fuzzed;
-  do {
-    generatedData = {};
-    fuzzed = goals[mode].generate(generatedData);
-  } while (plainModes.includes(mode) && /[0-9][zZ]/.test(fuzzed));
-  try {
-    const parsed = ES[`ParseTemporal${mode}String`](fuzzed);
-    for (let prop of comparisonItems[mode]) {
-      let expected = generatedData[prop];
-      if (prop !== 'ianaName' && prop !== 'offset' && prop !== 'calendar') expected = expected || 0;
-      assert.equal(parsed[prop], expected);
-    }
-    console.log(`${fuzzed} => ok`);
-  } catch (e) {
-    if (e instanceof assert.AssertionError) {
-      console.log(`${fuzzed} => parsed wrong: expected`, e.expected, 'actual', e.actual);
-      console.log('generated data:', generatedData);
-    } else {
-      console.log(`${fuzzed} failed!`, e);
+plainModes.forEach((mode) => {
+  console.log('// starting to fuzz ' + mode);
+  for (let count = 0; count < 1000; count++) {
+    let generatedData, fuzzed;
+    do {
+      generatedData = {};
+      fuzzed = goals[mode].generate(generatedData);
+    } while (/[0-9][zZ]/.test(fuzzed));
```

Comment:
I don't think this is correct. This skips testing Z strings for Instant and ZonedDateTime, where they are valid. (The purpose of this loop was to skip Z strings for Plain___ types, where they are not valid.) I think this loop can eventually be replaced by the early error mechanism that you're working on in #2036 but maybe it should be preserved for now.

Reply:
ah, I see, that make sense. I didn't understand that check.